### PR TITLE
Fix navigation text contrast

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

The navigation isn’t meeting contrast for the dark green #008060 text on the light grey selected background #EDEEEF. The contrast is currently 4.24 and should be 4.5.

### WHAT is this pull request doing?

Updating `polaris-tokens` to use the new `text-primary` color.

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [x] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Navigation/Navigation.scss [hunk 1]
```diff
@@ -104,7 +104,7 @@ $disabled-fade: 0.6;
       left: -1 * spacing(tight);
       height: 100%;
       width: $active-indicator-width;
-      background-color: var(--p-action-primary);
+      background-color: var(--p-text-primary);
```
Comment (aveline) [inline src/components/Navigation/Navigation.scss:107]: Should this be changed? 🤔

**Comment by kaelig** [comment]
Can you update https://github.com/Shopify/polaris-react/blob/master/documentation/Color%20system.md (`yarn run colordocs`) so it mentions the new token?

**Review by aveline — CHANGES_REQUESTED** [review]
L👀ks good except that one background colour change.

File: src/components/Navigation/Navigation.scss (hunk 1 above)
Comment (alex-page) [inline src/components/Navigation/Navigation.scss:107]:
```suggestion
      background-color: var(--p-action-primary);
```

Comment (alex-page) [reply src/components/Navigation/Navigation.scss]: Thanks @aveline ✨

**Pull request merged**